test/cloudant.test.js

```diff
@@ -102,61 +102,73 @@ describe('cloudant connector', function() {
       Product.destroyAll(null, {limit: testUtil.QUERY_MAX}, done);
     });
 
-    it('updates a single instance with array props',
-      function(done) {
-        prod1.setAttribute('type', ['cinnamon raisin']);
-        prod1.setAttribute('releases', [4, 5, 6]);
-        prod1.setAttribute('foo', [{id: 3, name: 'bread3'}]);
-        Product.updateAll({id: 1}, prod1, function(err, res) {
-          if (err) return done(err);
-          Product.findById('1', function(err, res) {
+    it('updates a single instance with array props', function(done) {
+      prod1.setAttribute('type', ['cinnamon raisin']);
+      prod1.setAttribute('releases', [4, 5, 6]);
+      prod1.setAttribute('foo', [{id: 3, name: 'bread3'}]);
+      Product.updateAll({id: 1}, prod1, function(err, res) {
+        if (err) return done(err);
+        Product.findById('1', function(err, res) {
+          if (err) done(err);
+          res.name.should.equal(prod1.name);
+          res.price.should.equal(prod1.price);
+          res.releases.should.deepEqual([4, 5, 6]);
+          res.type.should.deepEqual(['cinnamon raisin']);
+          res.foo.should.deepEqual([{id: 3, name: 'bread3'}]);
+          Product.findById('2', function(err, res) {
+            if (err) done(err);
+            res.name.should.equal(prod2.name);
+            res.price.should.equal(prod2.price);
+            res.releases.should.deepEqual(prod2.releases);
+            res.type.should.deepEqual(prod2.type);
+            res.foo.should.deepEqual(prod2.foo);
+            done();
+          });
+        });
+      });
+    });
+
+    it('updates all matching instances with array props', function(done) {
+      var data = {
+        price: 200,
+        releases: [7],
+        type: ['everything'],
+        foo: [{id: 1, name: 'bar'}],
+      };
+      Product.find(function(err, result) {
+        if (err) done(err);
+        var id1 = result[0].id;
```

Comment:
The usage of `result` here is a little dangerous...how do we make sure there are two items in the result :)

maybe `result.should.be.an.Array();` and `result.length.should.equal(2)`